analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,164 @@
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using SonarAnalyzer.SymbolicExecution;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution
+{
+    public static class SymbolicValueExtensions
+    {
+        public static SymbolicValueAssertions Should(this SymbolicValue instance) =>
+            new(instance);
+    }
+
+    public class SymbolicValueAssertions : ReferenceTypeAssertions<SymbolicValue, SymbolicValueAssertions>
+    {
+        protected override string Identifier => "symbolicValue";
+
+        public SymbolicValueAssertions(SymbolicValue subject) : base(subject) { }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveConstraint(SymbolicConstraint constraint, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("The {context:symbolicValue} is not present and can not have constraint {0}.", constraint)
+                .Then
+                .Given(() => Subject.AllConstraints.ToList())
+                .ForCondition(allConstraints => allConstraints.Count != 0)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has no constraints.", constraint)
+                .Then
+                .ForCondition(allConstraints => allConstraints.Contains(constraint))
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has {1} constraints.",
+                    _ => constraint,
+                    x => x.OrderBy(x => x.ToString()));
+            var matchedConstrainted = Subject.AllConstraints.First(x => x == constraint);
+            return new(this, matchedConstrainted);
+        }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveOnlyConstraint(SymbolicConstraint constraint, string because = "", params object[] becauseArgs)
```

Comment:
Let's use every single method at least once, somewhere in the existing UTs. To prove the point and to demonstrate the difference

Reply:
Line 72 `public AndConstraint<SymbolicValueAssertions> HaveOnlyConstraints(params SymbolicConstraint[] expected)` is not used

Reply:
The other overload is called here https://github.com/SonarSource/sonar-dotnet/pull/6034/files#diff-55030245a085cb0374d3ba253a79c91d4949adcb8b05bb98736978a8414ad04fR154 It illustrates the need for both overloads in case you want to give a `because` in the assertion.